src/main/scala/cromwell/binding/Scope.scala

```diff
@@ -106,9 +129,17 @@ object Scope {
   }
 
   @tailrec
-  def fullyQualifiedNameBuilder(scope: Option[Scope], fqn: String, fullDisplay: Boolean): String = scope match {
-    case None => fqn.tail //Strip away the first "." of the name
-    case Some(x: Scope) => fullyQualifiedNameBuilder(x.parent, (if (fullDisplay || x.appearsInFQN) s".${x.name}" else "") + fqn, fullDisplay)
+  def fullyQualifiedNameBuilder(scope: Option[Scope], fqn: String, fullDisplay: Boolean, leaf: Boolean): String = {
+    scope match {
+      case None =>
+        fqn.tail //Strip away the first "." of the name
```

Comment:
yeah i don't feel super strongly about it, and I've done it the way you describe for exactly that reason (and I think this is pretty close to being exactly one of those). This was mainly a sanity check Q to make sure it seems defensible to you. OTOH I notice now that it came from the previous one anyways so *shrug*